neuraxle/metaopt/random.py

```diff
@@ -114,6 +117,278 @@ def _split(self, data_inputs):
+        :return: validation_data_inputs, validation_expected_outputs
+        """
+        splitted_data_inputs = self._validation_split(data_inputs)
+        if expected_outputs is not None:
+            splitted_expected_outputs = self._validation_split(expected_outputs)
+            return splitted_data_inputs, splitted_expected_outputs
+        return splitted_data_inputs
+
+    def _train_split(self, data_inputs):
+        splitted_data_inputs = []
+        number_step = self._get_number_fold(data_inputs)
+
+        for i in range(number_step):
+            # first slice index is always 0 for walk forward cross validation
+            a = int(i * self.validation_window_size)
+            b = int(a + self.training_window_size)
+
+            if b > data_inputs.shape[1]:
+                b = data_inputs.shape[1]
+
+            # TODO: The slicer could a inverse_transform of the joiner. A len method should also be defined.
+            slice = data_inputs[:, a:b]
+            splitted_data_inputs.append(slice)
+        return splitted_data_inputs
+
+    def _validation_split(self, data_inputs):
+        splitted_data_inputs = []
+        number_step = self._get_number_fold(data_inputs)
+        for i in range(number_step):
+            a = int(self._validation_initial_start + i * self.validation_window_size)
+            b = int(a + self.validation_window_size)
+
+            if b > data_inputs.shape[1]:
+                b = data_inputs.shape[1]
+
+            # TODO: The slicer could a inverse_transform of the joiner. A len method should also be defined.
+            slice = data_inputs[:, a:b]
+            splitted_data_inputs.append(slice)
+        return splitted_data_inputs
+
+    def _get_number_fold(self, data_inputs):
+        if self.drop_remainder:
+            number_step = math.floor(
+                (data_inputs.shape[1] - self._validation_initial_start) / float(self.validation_window_size)
+            )
+        else:
+            number_step = math.ceil(
+                (data_inputs.shape[1] - self._validation_initial_start) / float(self.validation_window_size)
+            )
+        return number_step
+
+
+class AnchoredWalkForwardTimeSeriesCrossValidation(BaseCrossValidation):
+    """
+    This will change perform a anchored walk forward cross validation by performing a forward rolling split.
+    All the training split will start at the start of the time series but the end will increase toward
+    the end at each forward split. For the validation split it will start after a certain time delay (if padding is set)
+    after their corresponding training split.
+
+    The size of the validation split is defined by `validation_window_size`.
+    The difference in start position between two consecutive validation split is also equal to `validation_window_size`.
+
+    Notes: The data supported by this cross validation is nd.array of shape = (batch, timeSeries, ...) .
```

Comment:
`time_step` or `total_time_steps` would be more clear and consistent than `timeSeries` perhaps. And I prefer `batch_size` as it's a number instead of an actual array. And by convention I've always used square `[` and `]` brackets when it comes the time to do comment about the dimensions. For exemple, two lines below, you wrote `[batch_size, total_time_steps, n_features]`, that's perfect like that.

Reply:
Will change it in this PR.